crates/telio-pmtu/src/linux.rs

```diff
@@ -0,0 +1,378 @@
+use std::{ffi::c_int, io, net::IpAddr, os::fd::AsRawFd, time::Duration};
+
+use pnet_packet::{icmp, icmpv6, Packet};
+use telio_utils::telio_log_debug;
+use tokio::net::UdpSocket;
+
+pub struct PMTUSocket {
+    sock: UdpSocket,
+    host: IpAddr,
+}
+
+const TIMEOUT: Duration = Duration::from_secs(1);
```

Comment:
IMHO, I would recommend bumping this to 5s.

Reply:
Done